### DISCUSSION TITLE
Add sidebar icon scaling option

### DISCUSSION THREAD

**Discussion by raysdigitalfootprint** [runelite/runelite#14945] [Ideas]
Please add an option to the general RuneLite settings for scaling the side bar icona up and even showing a previous of the plugin name. I have so many plugins installed that it has become difficult to identify the plugin I am looking for based on the small icon image. 

I play approximately 3 hours a week on my main and GIM account and I cannot always remember what the small image/icon of eg "Supplies tracker",  "Loot tracker", "Skills calculator" ect ect looks like. As a filthy casual it becomes difficult with so many plugins that use these little side bar / little icons.

**Comment by CalebWhiting** [comment]
I implemented this but honestly, I don't like it, the icons look kind of ridiculous next to the non-scaled components and as you can see, even with no plugins enabled, there are serious space constraints on the panel without some form of scrolling.
![image](https://user-images.githubusercontent.com/1300430/164995370-89edca9c-3869-41cd-b24a-aa200ced4627.png)

**Reply by youriyep** [reply]
Could this please be implemented? The default icons are just so tiny for me it is pretty much unusable. Using --scale 2 makes runelite sidebar infinitely better for me but the runescape ui itself too massive, the inventory covers the minimap etc.

**Comment by Enriath** [comment]
If you're having trouble seeing things on the sidebar, the better option would be to use the `--scale` option with the launcher
e.g.
```
"%localappdata%\runelite\runelite.exe" --scale=2
```

**Comment by CraigBanach** [comment]
> If you're having trouble seeing things on the sidebar, the better option would be to use the `--scale` option with the launcher e.g.
> 
> ```
> "%localappdata%\runelite\runelite.exe" --scale=2
> ```

This worked a treat for me on a 4k monitor, perhaps it's worthwhile adding somewhere more accessible?

**Comment by CalebWhiting** [comment]
> > If you're having trouble seeing things on the sidebar, the better option would be to use the `--scale` option with the launcher e.g.
> > ```
> > "%localappdata%\runelite\runelite.exe" --scale=2
> > ```
> 
> This worked a treat for me on a 4k monitor, perhaps it's worthwhile adding somewhere more accessible?

I haven't used Windows in years so my memory may be failing me, but I think it's pretty easy to just right-click "Properties" on the RuneLite shortcut and add the argument?

**Comment by robmonte** [comment]
I wanted to bump this topic up because of the (relatively) recent update that makes the sidebar add more columns when the amount of icons exceeds the display's vertical space. I think adding scaling for these icons is a huge accessibility benefit. I personally have an _extremely_ hard time seeing individual icons to determine which plugins I am opening.

I do not find the `--scale` flag alternative to be a very great solution because it scales everything in Runelite including the game UI itself, all of which are most things I do not want scaled. It is just the tiny sidebar icons I have such struggles with.

In @CalebWhiting 's example implementation the concern was large icons running out of space in the panel and that is now no longer a concern, so hopefully it's up for consideration again. An additional benefit of supporting this now _could_ be Runelite supporting the option for a higher resolution icon instead of simply scaling the tiny one up, though of course existing plugins would have to then provide such an image first, if it's not already larger.

**Comment by roopeky** [comment]
uppppppppp

**Comment by Sigmonster** [comment]
bump,  I just need larger buttons so I can easily click them. UI scaling doesn't not have the desired effect, it makes the whole side bar much larger.

I don't care about the visual aspect, maybe make the icons slightly smaller but make the the clickable buttons larger. Also, can you maybe make this a settings options or something?

**Comment by Marwerk** [comment]
Please, this is needed so badly for people with visibility issues or wrist issues (makes clicking small things harder and more effort).

thank you for all of your great and hard work y'all, this client is gamechanging

**Comment by ArjG** [comment]
Can you make the scale  a float value please? I find a 2 scale to be too large, but a 1 scale to be too small. 1.5 would be perfect, but it doesnt work

**Reply by JZomDev** [reply]
This is because your windows manager isn't allowing it.
